apm-agent-plugins/apm-grpc/apm-grpc-plugin/src/main/java/co/elastic/apm/agent/grpc/GrpcHelper.java

```diff
@@ -165,9 +166,32 @@ public void setTransactionStatus(Status status, @Nullable Throwable thrown, Serv
         }
     }
 
+    public static Outcome toOutcome(@Nullable Status status){
+        Outcome outcome = Outcome.UNKNOWN;
+
+        if (status != null) {
+            switch (status.getCode()) {
+                case OK:
+                    outcome = Outcome.SUCCESS;
+                    break;
+                case UNKNOWN:
+                    outcome = Outcome.UNKNOWN;
```

Comment:
The `io.grpc.Status.Code#UNKNOWN` javadoc says this indicates an unknown **error**, so the outcome should be `FAILURE` as well.
I think that in general, if we know how to identify client failures, anything else should be considered as a success.